src/utils/rfqm_health_check.ts

```diff
@@ -76,3 +86,71 @@ export async function checkSqsQueueAsync(producer: Producer): Promise<HealthChec
     }
     return results;
 }
+
+/**
+ * Looks at the worker heartbeats and produces appropriate issues based on the age
+ * of the heartbeats and the worker balances.
+ *
+ * Heartbeat Age: Checks the most recent heartbeat and produces a FAILED issue if it is older than the failed
+ * threshold. For heartbeats other than the most recent, will only produce a DEGRADED issue. (i.e. the check only
+ * fails if ALL workers are stuck)
+ *
+ * Worker Balance: Like with the age check, this only produces a FAILED issue if all workers are below the failed
+ * balance. Individual worker balances produce a DEGRADED issue if they are below BALANCE_DEGRADED_THRESHOLD.
+ *
+ * Current date is an optional parameter for testing.
+ */
+export async function checkWorkerHeartbeatsAsync(
+    heartbeats: RfqmWorkerHeartbeatEntity[],
+    nowDate: Date = new Date(),
+): Promise<HealthCheckIssue[]> {
+    const results: HealthCheckIssue[] = [];
+    if (!heartbeats.length) {
+        return [{ status: HealthCheckStatus.Failed, description: 'No worker heartbeats were found' }];
+    }
+
+    // Heartbeat Age
+    const sortedHeartbeats = heartbeats.sort((a, b) => b.timestamp.getTime() - a.timestamp.getTime());
+    const newestHeartbeat = sortedHeartbeats[0];
+    const newestHeartbeatAgeMinutes = (nowDate.getTime() - newestHeartbeat.timestamp.getTime()) / MS_IN_MINUTE;
+    if (newestHeartbeatAgeMinutes > RECENT_HEARTBEAT_AGE_THRESHOLD) {
+        results.push({
+            status: HealthCheckStatus.Failed,
+            description: `No worker has published a heartbeat in the last ${RECENT_HEARTBEAT_AGE_THRESHOLD} minutes`,
+        });
+    }
+    // TODO (rhinodavid): Think about how this will work when we downscale and a worker isn't producing new
+    // hearbeats because it's been removed.
+    sortedHeartbeats.forEach(({ index, timestamp, address }) => {
+        const heartbeatAgeMinutes = (nowDate.getTime() - timestamp.getTime()) / MS_IN_MINUTE;
+        if (heartbeatAgeMinutes > RECENT_HEARTBEAT_AGE_THRESHOLD) {
```

Comment:
```suggestion
        if (heartbeatAgeMinutes >= RECENT_HEARTBEAT_AGE_THRESHOLD) {
```

tiny nit but I think of meeting the threshold as counting

Reply:
done